ios/Exponent/Kernel/Views/EXAppViewController.m

```diff
@@ -315,6 +318,34 @@ - (void)errorViewDidSelectRetry:(EXErrorView *)errorView
 
 - (UIInterfaceOrientationMask)supportedInterfaceOrientations
 {
+  NSString *validatedVersion = [[EXVersions sharedInstance] availableSdkVersionForManifest:_appRecord.appLoader.manifest];
+  NSString *majorSDKVersion = [validatedVersion componentsSeparatedByString:@"."][0];
+  if ([validatedVersion length] != 0 && [majorSDKVersion integerValue] <= 35) {
+    return [self legacySupportedInterfaceOrientations];
+  }
+  
+  if (_appRecord.experienceId == nil) {
+    return UIInterfaceOrientationMaskAllButUpsideDown;
+  }
+  EXScreenOrientationRegistry *registry = (EXScreenOrientationRegistry *)[UMModuleRegistryProvider getSingletonModuleForClass:[EXScreenOrientationRegistry class]];
+  if ([registry doesKeyExistForAppId:_appRecord.experienceId]) {
```

Comment:
Let's return `UIInterfaceOrientationUnknown` from `orientationMaskForAppId` in case experience hasn't defined the orientation mask (remove method `doesKeyExistForAppId` altogether, we shouldn't need to ask whether we can ask — let's go straight to asking)

Reply:
We want to know if the key exists, because if not, it means that user didn't modify orientation mask and then we look into manifest (else if). After user did something with orientation mask, we don't want to check manifest anymore. Removing doesKeyExistForAppId will leave us without this information. If I can remember correctly, that was the reasoning behind this method.

Reply:
Would there be a problem if we said that not setting or setting orientation to `UNKNOWN` means _please read from manifest_? This has two advantages:
- removes artificial `doesKeyExistForAppId` method
- allows user to fallback back to manifest setting.